README.md

```diff
@@ -42,54 +64,176 @@ public interface JpaCqlRepository<T, ID> extends JpaRepository<T, ID> {
+| `public`    | `loadSampleData()`                        | Load any sample data (requested with `loadSample=true` parameter)                               |
+| `protected` | `beforeTenantUpdate(TenantAttributes)`    | Run custom logic before a tenant is created or updated                                          |
+| `protected` | `beforeLiquibaseUpdate(TenantAttributes)` | Run custom logic immediately before Liquibase updates are started (after `beforeTenantUpdate`)  |
+| `protected` | `afterLiquibaseUpdate(TenantAttributes)`  | Run custom logic immediately before Liquibase updates are finished (before `afterTenantUpdate`) |
+| `protected` | `afterTenantUpdate(TenantAttributes)`     | Run custom logic after all update jobs are completed                                            |
+| `protected` | `beforeTenantDeletion(TenantAttributes)`  | Run custom logic before a tenant is deleted/purged                                              |
+| `protected` | `afterTenantDeletion(TenantAttributes)`   | Run custom logic after a tenant is deleted/purged (the schema will no longer exist)             |
+
+### `TenantService` Methods and Fields
+
+There are two methods that may be of use in your custom logic:
+
+- `boolean tenantExists()` which will check if the database schema for this tenant exists (this
+  says nothing about if it is up to date)
+- `String getSchemaName()` will construct and return the name of the schema corresponding to the
+  module and tenant
+
+These fields will also be provided:
+
+- `JdbcTemplate jdbcTemplate`, for running Postgres queries directly
+- `FolioExecutionContext context`, for getting information about the module
+- `FolioSpringLiquibase folioSpringLiquibase`, for interacting with Liquibase directly (this
+  extends `SpringLiquibase` and may be `null` if Liquibase is not enabled!)
+
+### Event Order
+
+The [events](#tenantservice-event-methods) will be called in the following order:
+
+#### Upon Creation
+
+1. `beforeTenantUpdate`
+2. If Liquibase is enabled:
+   1. `beforeLiquibaseUpdate`
+   2. _Internal logic to apply Liquibase changes_
+   3. `afterLiquibaseUpdate`
+3. `afterTenantUpdate`
+4. `loadReferenceData`, if applicable
+5. `loadSampleData`, if applicable
+
+#### Upon Deletion
+
+1. `beforeTenantDeletion`
+2. _Internal logic to drop the schema_
+3. `afterTenantDeletion`
+
+### Sample
+
+Overriding these methods to add your own custom logic is quite straightforward. Here is an example
+of how to override these in your very own `@Service`:
+
+```java
+package org.folio.yourmodule.service;
+
+import javax.annotation.Priority;
+import org.folio.spring.service.TenantService;
+import org.folio.tenant.domain.dto.TenantAttributes;
+import org.folio.yourmodule.SuperCoolDataRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Priority(-10) // required to ensure CustomTenantService will be loaded instead of TenantService
```

Comment:
Just a note:
I'd probably got with `@Primary` annotation instead. These shouldn't be more than one implementation of TenantService, so using `Priority` or `Order` that assume several options in general is less appropriate here in my opinion, but Priority will also work.